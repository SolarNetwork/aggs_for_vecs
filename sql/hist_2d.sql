-- smallint hist_2d
SELECT hist_2d(smallints[1], smallints[2], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements;
-- smallint hist_2d none
SELECT hist_2d(smallints[1], smallints[2], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements WHERE sensor_id = -1;
-- smallint hist_2d one null
SELECT hist_2d(smallints[1], smallints[2], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements WHERE sensor_id = 1;
-- smallint hist_2d array of nulls
SELECT hist_2d(smallints[1], smallints[2], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements WHERE sensor_id = 2;
-- smallint hist_2d one not-null
SELECT hist_2d(smallints[1], smallints[3], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements WHERE sensor_id = 4;
-- smallint hist_2d array of nulls and one other
SELECT hist_2d(smallints[1], smallints[3], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements WHERE sensor_id IN (2, 4);

-- int hist_2d
SELECT hist_2d(ints[1], ints[2], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements;
-- int hist_2d none
SELECT hist_2d(ints[1], ints[2], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements WHERE sensor_id = -1;
-- int hist_2d one null
SELECT hist_2d(ints[1], ints[2], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements WHERE sensor_id = 1;
-- int hist_2d array of nulls
SELECT hist_2d(ints[1], ints[2], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements WHERE sensor_id = 2;
-- int hist_2d one not-null
SELECT hist_2d(ints[1], ints[3], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements WHERE sensor_id = 4;
-- int hist_2d array of nulls and one other
SELECT hist_2d(ints[1], ints[3], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements WHERE sensor_id IN (2, 4);

-- bigint hist_2d
SELECT hist_2d(bigints[1], bigints[2], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements;
-- bigint hist_2d none
SELECT hist_2d(bigints[1], bigints[2], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements WHERE sensor_id = -1;
-- bigint hist_2d one null
SELECT hist_2d(bigints[1], bigints[2], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements WHERE sensor_id = 1;
-- bigint hist_2d array of nulls
SELECT hist_2d(bigints[1], bigints[2], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements WHERE sensor_id = 2;
-- bigint hist_2d one not-null
SELECT hist_2d(bigints[1], bigints[3], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements WHERE sensor_id = 4;
-- bigint hist_2d array of nulls and one other
SELECT hist_2d(bigints[1], bigints[3], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements WHERE sensor_id IN (2, 4);

-- real hist_2d
SELECT hist_2d(reals[1], reals[2], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements;
-- real hist_2d none
SELECT hist_2d(reals[1], reals[2], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements WHERE sensor_id = -1;
-- real hist_2d one null
SELECT hist_2d(reals[1], reals[2], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements WHERE sensor_id = 1;
-- real hist_2d array of nulls
SELECT hist_2d(reals[1], reals[2], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements WHERE sensor_id = 2;
-- real hist_2d one not-null
SELECT hist_2d(reals[1], reals[3], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements WHERE sensor_id = 4;
-- real hist_2d array of nulls and one other
SELECT hist_2d(reals[1], reals[3], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements WHERE sensor_id IN (2, 4);

-- float8 hist_2d
SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements;
-- float8 hist_2d none
SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = -1;
-- float8 hist_2d one null
SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = 1;
-- float8 hist_2d array of nulls
SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = 2;
-- float8 hist_2d one not-null
SELECT hist_2d(floats[1], floats[3], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = 4;
-- float8 hist_2d array of nulls and one other
SELECT hist_2d(floats[1], floats[3], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id IN (2, 4);
