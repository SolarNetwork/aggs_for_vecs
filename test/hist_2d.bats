load test_helper

@test "int16 hist_2d" {
  result="$(query "SELECT hist_2d(ints[1], ints[2], 0, 0, 1, 1, 3, 3) FROM measurements")";
  echo $result;
  [ "$result" = "{{0,0,0},{0,1,0},{0,2,0}}" ]
}

# @test "int32 hist_2d" {
  # result="$(query "SELECT hist_2d('{1,1,5,2,0}'::integer[])")";
  # [ "$result" = "0" ]
# }

# @test "int64 hist_2d" {
  # result="$(query "SELECT hist_2d('{1,1,5,2,0}'::bigint[])")";
  # [ "$result" = "0" ]
# }

# @test "float4 hist_2d" {
  # result="$(query "SELECT hist_2d('{1,1,5,2,0}'::real[])")";
  # [ "$result" = "0" ]
# }

@test "float8 hist_2d" {
  result="$(query "SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements")";
  echo $result;
  [ "$result" = "{{0,0,0},{0,1,0},{0,2,0}}" ]
}

@test "float8 hist_2d none" {
  result="$(query "SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = -1")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "float8 hist_2d one null" {
  result="$(query "SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = 1")";
  echo $result;
  [ "$result" = "{{0,0,0},{0,0,0},{0,0,0}}" ]
}

@test "float8 hist_2d array of nulls" {
  result="$(query "SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = 2")";
  echo $result;
  [ "$result" = "{{0,0,0},{0,0,0},{0,0,0}}" ]
}

@test "float8 hist_2d one not-null" {
  result="$(query "SELECT hist_2d(floats[1], floats[3], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = 4")";
  echo $result;
  [ "$result" = "{{0,0,0},{0,0,0},{0,1,0}}" ]
}

@test "float8 hist_2d array of nulls and one other" {
  result="$(query "SELECT hist_2d(floats[1], floats[3], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id IN (2, 4)")";
  echo $result;
  [ "$result" = "{{0,0,0},{0,0,0},{0,1,0}}" ]
}
