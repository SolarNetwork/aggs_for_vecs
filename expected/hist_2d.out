-- smallint hist_2d
SELECT hist_2d(smallints[1], smallints[2], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements;
          hist_2d          
---------------------------
 {{0,0,0},{0,1,0},{0,2,0}}
(1 row)

-- smallint hist_2d none
SELECT hist_2d(smallints[1], smallints[2], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements WHERE sensor_id = -1;
 hist_2d 
---------
 
(1 row)

-- smallint hist_2d one null
SELECT hist_2d(smallints[1], smallints[2], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements WHERE sensor_id = 1;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,0,0}}
(1 row)

-- smallint hist_2d array of nulls
SELECT hist_2d(smallints[1], smallints[2], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements WHERE sensor_id = 2;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,0,0}}
(1 row)

-- smallint hist_2d one not-null
SELECT hist_2d(smallints[1], smallints[3], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements WHERE sensor_id = 4;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,1,0}}
(1 row)

-- smallint hist_2d array of nulls and one other
SELECT hist_2d(smallints[1], smallints[3], 0::smallint, 0::smallint, 1::smallint, 1::smallint, 3, 3) FROM measurements WHERE sensor_id IN (2, 4);
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,1,0}}
(1 row)

-- int hist_2d
SELECT hist_2d(ints[1], ints[2], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements;
          hist_2d          
---------------------------
 {{0,0,0},{0,1,0},{0,2,0}}
(1 row)

-- int hist_2d none
SELECT hist_2d(ints[1], ints[2], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements WHERE sensor_id = -1;
 hist_2d 
---------
 
(1 row)

-- int hist_2d one null
SELECT hist_2d(ints[1], ints[2], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements WHERE sensor_id = 1;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,0,0}}
(1 row)

-- int hist_2d array of nulls
SELECT hist_2d(ints[1], ints[2], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements WHERE sensor_id = 2;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,0,0}}
(1 row)

-- int hist_2d one not-null
SELECT hist_2d(ints[1], ints[3], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements WHERE sensor_id = 4;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,1,0}}
(1 row)

-- int hist_2d array of nulls and one other
SELECT hist_2d(ints[1], ints[3], 0::int, 0::int, 1::int, 1::int, 3, 3) FROM measurements WHERE sensor_id IN (2, 4);
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,1,0}}
(1 row)

-- bigint hist_2d
SELECT hist_2d(bigints[1], bigints[2], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements;
          hist_2d          
---------------------------
 {{0,0,0},{0,1,0},{0,2,0}}
(1 row)

-- bigint hist_2d none
SELECT hist_2d(bigints[1], bigints[2], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements WHERE sensor_id = -1;
 hist_2d 
---------
 
(1 row)

-- bigint hist_2d one null
SELECT hist_2d(bigints[1], bigints[2], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements WHERE sensor_id = 1;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,0,0}}
(1 row)

-- bigint hist_2d array of nulls
SELECT hist_2d(bigints[1], bigints[2], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements WHERE sensor_id = 2;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,0,0}}
(1 row)

-- bigint hist_2d one not-null
SELECT hist_2d(bigints[1], bigints[3], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements WHERE sensor_id = 4;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,1,0}}
(1 row)

-- bigint hist_2d array of nulls and one other
SELECT hist_2d(bigints[1], bigints[3], 0::bigint, 0::bigint, 1::bigint, 1::bigint, 3, 3) FROM measurements WHERE sensor_id IN (2, 4);
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,1,0}}
(1 row)

-- real hist_2d
SELECT hist_2d(reals[1], reals[2], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements;
          hist_2d          
---------------------------
 {{0,0,0},{0,1,0},{0,2,0}}
(1 row)

-- real hist_2d none
SELECT hist_2d(reals[1], reals[2], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements WHERE sensor_id = -1;
 hist_2d 
---------
 
(1 row)

-- real hist_2d one null
SELECT hist_2d(reals[1], reals[2], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements WHERE sensor_id = 1;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,0,0}}
(1 row)

-- real hist_2d array of nulls
SELECT hist_2d(reals[1], reals[2], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements WHERE sensor_id = 2;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,0,0}}
(1 row)

-- real hist_2d one not-null
SELECT hist_2d(reals[1], reals[3], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements WHERE sensor_id = 4;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,1,0}}
(1 row)

-- real hist_2d array of nulls and one other
SELECT hist_2d(reals[1], reals[3], 0::real, 0::real, 1::real, 1::real, 3, 3) FROM measurements WHERE sensor_id IN (2, 4);
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,1,0}}
(1 row)

-- float8 hist_2d
SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements;
          hist_2d          
---------------------------
 {{0,0,0},{0,1,0},{0,2,0}}
(1 row)

-- float8 hist_2d none
SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = -1;
 hist_2d 
---------
 
(1 row)

-- float8 hist_2d one null
SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = 1;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,0,0}}
(1 row)

-- float8 hist_2d array of nulls
SELECT hist_2d(floats[1], floats[2], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = 2;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,0,0}}
(1 row)

-- float8 hist_2d one not-null
SELECT hist_2d(floats[1], floats[3], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id = 4;
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,1,0}}
(1 row)

-- float8 hist_2d array of nulls and one other
SELECT hist_2d(floats[1], floats[3], 0::float, 0::float, 1::float, 1::float, 3, 3) FROM measurements WHERE sensor_id IN (2, 4);
          hist_2d          
---------------------------
 {{0,0,0},{0,0,0},{0,1,0}}
(1 row)

